### ISSUE TITLE
Pushed view props are not updated when changed on parent view

### ISSUE THREAD

**Issue by MarlBurroW** [raycast/extensions#8888]
### Description

When you push a view with a prop as shown below:

```jsx
 <Action.Push title="Post Detail" target={<Post post={post} />} />
 ```
It appears that the post prop becomes immutable in the child view, meaning any updates to the state in the parent view don't get passed to the child view's props.

While this might not pose a problem in many scenarios, if you're refreshing a list of items on the parent page (whether through websocket or polling), it would be intuitive to expect those updates to be reflected in the child view, especially if you're actively viewing it.
 
 Related issue: https://github.com/raycast/extensions/issues/896 (Closed)
 
 Current workaround I've found: Use a store library like zustand to share the state between parent and child view.

### Steps To Reproduce

I've tried to write a straigthforward example to explain the issue:

```tsx
import { Action, ActionPanel, List, Detail} from "@raycast/api";
import { useEffect, useState } from "react";

export default function Command() {

  const [counter, setCounter] = useState<number>(0);

  useEffect(() => {
   // Start updating the counter state every seconds
    const interval = setInterval(() => {
      setCounter(counter + 1);
    }, 1000);

    return () => clearInterval(interval);

  }, []);

  return (
    <List>
        <List.Item
          title={`Counter state: ${counter}`}
          actions={
            <ActionPanel>
              <Action.Push title="ChildPage" target={<ChildPage counter={counter} />} />
            </ActionPanel>
          }
        />
    </List>
  );
}

interface ChildPageProps {
  counter: number;
}
function ChildPage({counter}: ChildPageProps) {
  return (<Detail>{counter}</Detail>);
}
```

### Current Behaviour

Counter is not incremented when you are in the child view.
You have to go back to the parent view and re-click on the push action to view the new incremented value on the child view.

### Expected Behaviour

Counter should be incremented each seconds when you are in the child view.

**Comment by mathieudutour** [comment]
That's not a bug, it's not a "child" view but a "sibling" view. When you update a component, its siblings aren't updated.

But I agree that the `<Action.Push />` semantic is hiding a bit what's happening. It's more intuitive if you think of it as
```js
const {push} = useNavigation()

...

<Action title="ChildPage" onAction={() => push(<ChildPage counter={counter} />)} />
```

**Comment by MarlBurroW** [comment]
You're right, it's more like a "Go to this page" than "Open this page/modal on top of this one".
There is no page hierarchy, just top level pages that can be pushed to the raycast internal router.
I mistakingly thought that all pages pushed by a root command components were a kind of child view of this root command.

With this point of view, my workaround (using a store to share data between pages) appears to be the best solution to handle this kind of edge cases right ?

**Comment by mathieudutour** [comment]
Indeed, that works yes 👍

**Comment by MarlBurroW** [comment]
Ok thank you for enlightening me on this.
This is obviously not a bug and can be closed I guess.
